sql/expression/function/time_test.go

```diff
@@ -383,3 +404,43 @@ func TestDate(t *testing.T) {
 		})
 	}
 }
+
+func TestTimeDiff(t *testing.T) {
+	require := require.New(t)
+
+	date := time.Date(2018, time.December, 2, 16, 25, 0, 0, time.Local)
+	testNowFunc := func() time.Time {
+		return date
+	}
+
+	var ctx *sql.Context
+	err := sql.RunWithNowFunc(testNowFunc, func() error {
+		ctx = sql.NewEmptyContext()
+		return nil
+	})
+	require.NoError(err)
+
+	from := time.Date(2018, time.May, 2, 0, 0, 0, 0, time.Local)
+	to := time.Date(2018, time.May, 2, 0, 0, 1, 0, time.Local)
+	diff := NewTimeDiff(expression.NewLiteral(from, sql.Timestamp), expression.NewLiteral(to, sql.Timestamp))
+	result, err := diff.Eval(ctx, nil)
+	require.NoError(err)
+	expected := "-00:00:01"
+	require.Equal(expected, result)
+
+	from = time.Date(2008, time.December, 31, 23, 59, 59, 1, time.Local)
+	to = time.Date(2008, time.December, 30, 1, 1, 1, 2, time.Local)
+	diff = NewTimeDiff(expression.NewLiteral(from.UnixNano(), sql.Time), expression.NewLiteral(to.UnixNano(), sql.Time))
```

Comment:
`from.Unix()` and `to.Unix()` would break this i believe

Reply:
An easier way to do these would be to do something similar to how the types are tested. Have a struct array that is looped over, creating a timediff and calling eval on each item.